Client/Application/AppState.swift

```diff
@@ -8,7 +8,11 @@ protocol AppStateDelegate: class {
     func appDidUpdateState(appState: AppState)
 }
 
-enum AppState {
+struct AppState {
+    let ui: UIState
+}
+
+enum UIState {
```

Comment:
Not sure what retaining this enum gives us now. Can we not just make `UIState` a protocol and make `TabState`, `HomePanelState` and `TabTrayState` implement it?